### DISCUSSION TITLE
Wiznet

### DISCUSSION THREAD

**Discussion by thestumbler** [modm-io/modm#1086] [General]
I'm about to add the Wiznet 5100S to my project for UDP (TFTP). I think I've reviewed the issues and discussions here, and concluded the existing Ethernet examples don't apply -- they're all using the on-board MAC of the various STM32 flavors and a stack like FreeRTOS or maybe LwIP. Whereas the W5100s and cousins have a stack built in. 

My plan of attack is to use the existing SPI DMA module and port over the W5100s code already used by my main application (basically some reference code provided by WizNet). 

Am I on the right track, or have I missed a similar, existing project / example?

**Comment by salkinium** [comment]
That sounds about right. modm is not super strong on networking, we'd rather integrate third-party projects like FreeRTOS+TCP.

**Reply by thestumbler** [reply]
Yeah, I would love to do that as well. Alas, for reasons beyond my control, this hardware design uses the Wiznet chip and not the built-in Ethernet MAC. The Wiznet chip not only has the MAC, but the TCP/IP stack as well. I've not found any existing solution using FreeRTOS or LwIP, which I guess isn't surprising considering they are IP stacks and that isn't needed when you're using a Wiznet chip. 

I do plan to migrate future designs to use the internal Ethernet MAC, so I'll eventually incorporate FreeRTOS (probably that instead of LwIP, since my client is a big user of FreeRTOS in their embedded applications).

**Reply by thestumbler** [reply]
But on the other hand, it's not so bad, either. That Wiznet contains the stack, and MODM already provides an SPI module, the programming job should be straightforward. Famous last words.
